src/Microsoft.DotNet.XHarness.CLI/Commands/WebServer.cs

```diff
@@ -175,6 +176,19 @@ public void Configure(IApplicationBuilder app, IOptionsMonitor<TestWebServerOpti
                     });
                 });
             }
+            if (options.WebServerUploadResults)
+            {
+                app.UseRouter(router =>
+                {
+                    router.MapPost("/test-results", async context =>
+                    {
+                        var xmlResultsFilePath = Path.Combine(options.OutputDirectory ?? Directory.CreateTempSubdirectory().FullName, "testResults.xml");

```

Comment:
we shouldn't need to handle OutputDirectory being null since it's a required CLI option
```suggestion
                        var xmlResultsFilePath = Path.Combine(options.OutputDirectory, "testResults.xml");

```